### PR TITLE
Use context to fill in more details about error XML

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Use context to fill in more details about error XML

## Motivation and Context
Context-aware error

## Regression
No more like a feature

## How Has This Been Tested?
Using `mc` to reproduce different types of errors

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ebozduman** [comment]
@harshavardhana,
Please check the Travis failure

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#4990af0f-985f-4448-8c55-76dc2e0433db) |
<a name="4990af0f-985f-4448-8c55-76dc2e0433db"></a>
### 7232-6a883e0/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31541
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b13ad04520c9:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 2 seconds
(2/13) Running aws-sdk-java tests ... done in 4 seconds
(3/13) Running aws-sdk-php tests ... done in 47 seconds
(4/13) Running aws-sdk-ruby tests ... done in 15 seconds
(5/13) Running awscli tests ... done in 2 minutes and 2 seconds
(6/13) Running mc tests ... done in 31 seconds
(7/13) Running minio-dotnet tests ... done in 3 minutes and 55 seconds
(8/13) Running minio-go tests ... done in 1 minutes and 31 seconds
(9/13) Running minio-java tests ... FAILED in 25 seconds
{
  "name": "minio-java",
  "function": "putObject(String bucketName, String objectName, InputStream body, long size, String contentType)",
  "args": "contentType: application/javascript",
  "duration": 15050,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115), java.net.SocketOutputStream.write(SocketOutputStream.java:155), okio.Okio$1.write(Okio.java:78), okio.AsyncTimeout$1.write(AsyncTimeout.java:179), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:41), okhttp3.internal.http1.Http1Codec$FixedLengthSink.write(Http1Codec.java:286), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:109), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:74), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:62), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185), okhttp3.RealCall.execute(RealCall.java:69), io.minio.MinioClient.executeReq(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executePut(MinioClient.java:1399), io.minio.MinioClient.executePut(MinioClient.java:1421), io.minio.MinioClient.putObject(MinioClient.java:3986), io.minio.MinioClient.putObject(MinioClient.java:4078), io.minio.MinioClient.putObject(MinioClient.java:3952), FunctionalTest.testPutObjectUnknownStreamSize(FunctionalTest.java:556), FunctionalTest.putObject_test8(FunctionalTest.java:582), FunctionalTest.runTests(FunctionalTest.java:2998), FunctionalTest.main(FunctionalTest.java:3154)]"
}

Executed 8 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#e302d7b7-b3db-44d7-96db-dedd547e070b) |
<a name="e302d7b7-b3db-44d7-96db-dedd547e070b"></a>
### 7232-a576bbf/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30247
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dedb2c618981:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 44 seconds
(4/13) Running aws-sdk-ruby tests ... done in 3 seconds
(5/13) Running awscli tests ... done in 1 minutes and 54 seconds
(6/13) Running mc tests ... done in 32 seconds
(7/13) Running minio-dotnet tests ... done in 3 minutes and 3 seconds
(8/13) Running minio-go tests ... done in 57 seconds
(9/13) Running minio-java tests ... FAILED in 26 seconds
{
  "name": "minio-java",
  "function": "putObject(String bucketName, String objectName, InputStream body, long size, String contentType)",
  "args": "contentType: application/javascript",
  "duration": 20183,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115), java.net.SocketOutputStream.write(SocketOutputStream.java:155), okio.Okio$1.write(Okio.java:78), okio.AsyncTimeout$1.write(AsyncTimeout.java:179), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:41), okhttp3.internal.http1.Http1Codec$FixedLengthSink.write(Http1Codec.java:286), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:109), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:74), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:62), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185), okhttp3.RealCall.execute(RealCall.java:69), io.minio.MinioClient.executeReq(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executePut(MinioClient.java:1399), io.minio.MinioClient.executePut(MinioClient.java:1421), io.minio.MinioClient.putObject(MinioClient.java:3986), io.minio.MinioClient.putObject(MinioClient.java:4078), io.minio.MinioClient.putObject(MinioClient.java:3952), FunctionalTest.testPutObjectUnknownStreamSize(FunctionalTest.java:556), FunctionalTest.putObject_test8(FunctionalTest.java:582), FunctionalTest.runTests(FunctionalTest.java:2998), FunctionalTest.main(FunctionalTest.java:3154)]"
}

Executed 8 out of 13 tests successfully.
```

File: cmd/object-handlers_test.go [hunk 1]
```diff
@@ -527,11 +540,28 @@ func testAPIGetObjectHandler(obj ObjectLayer, instanceType, bucketName string, a
...
+				t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType, testCase.expectedContent, string(actualContent))
+			}
+			continue
+		}
+
 		// Verify whether the bucket obtained object is same as the one created.
-		if !bytes.Equal(testCase.expectedContent, actualContent) {
-			t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType, testCase.expectedContent, string(actualContent))
+		actualError := &APIErrorResponse{}
+		if err = xml.Unmarshal(actualContent, actualError); err != nil {
```
Comment (ebozduman) [inline cmd/object-handlers_test.go:555]: The following code is repeated in several places (6) :
```
actualError := &APIErrorResponse{}
if err = xml.Unmarshal(actualContent, actualError); err != nil {
	t.Fatalf("Test %d: %s: Failed parsing response body: <ERROR> %v", i+1, instanceType, err)
}

if actualError.BucketName != testCase.bucketName {
	t.Fatalf("Test %d: %s: Unexpected bucket name, expected %s, got %s", i+1, instanceType,testCase.bucketName, actualError.BucketName)
}

if actualError.Key != testCase.objectName {
	t.Fatalf("Test %d: %s: Unexpected object name, expected %s, got %s", i+1, instanceType,testCase.objectName, actualError.Key)
}
```
Even the following code can be included although it differs a little bit
```
if rec.Code == http.StatusOK || rec.Code == http.StatusPartialContent {
	if !bytes.Equal(testCase.expectedContent, actualContent) {
		t.Errorf("Test %d: %s: Object content differs from expected value %s, got %s", i+1, instanceType,testCase.expectedContent, string(actualContent))
	}
	continue
}
```
Can you DRY it?
Here are the file names and line numbers where this code is repeated:

FileName | Line#s
--------------|---------
cmd/object-handlers_test.go | 554 ~ 565
   //                                         | 603 ~ 614
   //                                         | 2847 ~ 2858
cmd/test-utils_test.go
   //                                         | 1785 ~ 1796
   //                                         | 1846 ~ 1857
   //                                         | 1901 ~ 1912

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: cmd/object-handlers_test.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/object-handlers_test.go:555]: These tests will be removed @ebozduman in a separate run - this PR is simply to make that they don't fail.

Comment (harshavardhana) [reply cmd/object-handlers_test.go:555]: So for now this PR is primarily to make our error XML dynamic..

Comment (ebozduman) [reply cmd/object-handlers_test.go:555]: OK

**Review by ebozduman — DISMISSED** [review]
LGTM

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 7232-41d8384/gopath/src/github.com/minio
+ git -C 7232-41d8384/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 7232-41d8384/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 7232-41d8384/gopath/src/github.com/minio/minio fetch --quiet minio pull/7232/head:pr7232
+ git -C 7232-41d8384/gopath/src/github.com/minio/minio checkout --quiet pr7232
+ GOPATH=/mnt/nvme/mint-auto/7232-41d8384/gopath
+ make -C 7232-41d8384/gopath/src/github.com/minio/minio --quiet
# github.com/minio/minio/cmd
cmd/handler-utils.go:373:6: syntax error: unexpected notFoundHandler, expecting (
make: *** [build] Error 2

```

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 7232-9773697/gopath/src/github.com/minio
+ git -C 7232-9773697/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 7232-9773697/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 7232-9773697/gopath/src/github.com/minio/minio fetch --quiet minio pull/7232/head:pr7232
+ git -C 7232-9773697/gopath/src/github.com/minio/minio checkout --quiet pr7232
+ GOPATH=/mnt/nvme/mint-auto/7232-9773697/gopath
+ make -C 7232-9773697/gopath/src/github.com/minio/minio --quiet
# github.com/minio/minio/cmd
cmd/admin-handlers.go:449:25: not enough arguments in call to writeErrorResponseJSON
	have ("net/http".ResponseWriter, APIError, *url.URL)
	want (context.Context, "net/http".ResponseWriter, APIError, *url.URL)
make: *** [build] Error 2

```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_exclamation_mark: [more...](#9d86eaee-6ff0-4cda-a80f-9ffdeb54f9ca) |
| mint-xl.sh | :heavy_exclamation_mark: [more...](#10b43f96-a562-4010-9dcd-ab5de654a629) |
| mint-large-bucket.sh | :heavy_exclamation_mark: [more...](#40b75646-a658-42d8-bbfd-c98ae07d927e) |
| mint-compression-dist-xl.sh | :heavy_exclamation_mark: [more...](#11565ce8-0e2f-47ee-82bc-28dfea7a753d) |
| mint-compression-fs.sh | :heavy_exclamation_mark: [more...](#5eb5f128-fda8-4d39-a943-90cb71bad60b) |
| mint-worm.sh | :heavy_exclamation_mark: [more...](#e3e31bfd-8693-4182-b0b3-c4784807d0b3) |
| mint-fs.sh | :heavy_exclamation_mark: [more...](#a1e0a4d7-15d2-4435-a78a-65cc1226ac1d) |
| mint-dist-xl.sh | :heavy_exclamation_mark: [more...](#b2f8df12-8f91-4344-b189-80043bbf64f6) |
| mint-gateway-nas.sh | :heavy_exclamation_mark: [more...](#6c008780-656d-4e91-8fd4-b30ab76a914c) |
<a name="6c008780-656d-4e91-8fd4-b30ab76a914c"></a>
### 7232-49ff8c0/mint-gateway-nas.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="5eb5f128-fda8-4d39-a943-90cb71bad60b"></a>
### 7232-49ff8c0/mint-compression-fs.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="b2f8df12-8f91-4344-b189-80043bbf64f6"></a>
### 7232-49ff8c0/mint-dist-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="a1e0a4d7-15d2-4435-a78a-65cc1226ac1d"></a>
### 7232-49ff8c0/mint-fs.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="e3e31bfd-8693-4182-b0b3-c4784807d0b3"></a>
### 7232-49ff8c0/mint-worm.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="9d86eaee-6ff0-4cda-a80f-9ffdeb54f9ca"></a>
### 7232-49ff8c0/mint-compression-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="11565ce8-0e2f-47ee-82bc-28dfea7a753d"></a>
### 7232-49ff8c0/mint-compression-dist-xl.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="40b75646-a658-42d8-bbfd-c98ae07d927e"></a>
### 7232-49ff8c0/mint-large-bucket.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="10b43f96-a562-4010-9dcd-ab5de654a629"></a>
### 7232-49ff8c0/mint-xl.sh.xtrace:
```
DISABLED TIME BEING
```

**Review by ebozduman — APPROVED** [review]
LGTM

File: cmd/api-response.go [hunk 2]
```diff
@@ -596,25 +598,28 @@ func writeErrorResponseHeadersOnly(w http.ResponseWriter, err APIError) {
...
 	errorResponse := APIErrorResponse{
-		Code:      err.Code,
-		Message:   errBody,
-		Resource:  reqURL.Path,
-		RequestID: w.Header().Get(responseRequestIDKey),
-		HostID:    "3L137",
+		Code:       err.Code,
+		Message:    errBody,
+		Resource:   reqURL.Path,
+		BucketName: reqInfo.BucketName,
```
Comment (kannappanr) [inline cmd/api-response.go:619]: `reqInfo` can be `nil`

Comment (kannappanr) [reply cmd/api-response.go:619]: We need to set `BucketName`  and `ObjectName` only if it is not `nil`. do we not have this info already as part of `Resource`?

Comment (harshavardhana) [reply cmd/api-response.go:619]: Really when ? Isn't newContext support to always add reqInfo . If it's nil then it's a bug.

Comment (kannappanr) [reply cmd/api-response.go:619]: We are also using `context.Background()`

Comment (harshavardhana) [reply cmd/api-response.go:619]: Ah then yes..

Comment (kannappanr) [reply cmd/api-response.go:619]: I looked at `GetReqInfo` code. It returns, empty `ReqInfo` in that case, so we should be good.

Comment (kannappanr) [reply cmd/api-response.go:619]: `reqInfo` will only be `nil` if `ctx` is `nil`

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by ebozduman** [comment]
@harshavardhana,
How about `getSTSErrorResponse` just like `getAPIErrorResponse`?

**Comment by harshavardhana** [comment]
> How about `getSTSErrorResponse` just like `getAPIErrorResponse`?

Will do in next iteration.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#59e3050d-e4c7-408e-ae8d-8d096a9ec117) |
<a name="59e3050d-e4c7-408e-ae8d-8d096a9ec117"></a>
### 7232-49ff8c0/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30961
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 92d2ad007047:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 1 seconds
(2/13) Running aws-sdk-java tests ... done in 0 seconds
(3/13) Running aws-sdk-php tests ... done in 42 seconds
(4/13) Running aws-sdk-ruby tests ... done in 2 seconds
(5/13) Running awscli tests ... done in 1 minutes and 51 seconds
(6/13) Running mc tests ... done in 18 seconds
(7/13) Running minio-dotnet tests ... done in 2 minutes and 23 seconds
(8/13) Running minio-go tests ... done in 14 seconds
(9/13) Running minio-java tests ... FAILED in 11 seconds
{
  "name": "minio-java",
  "function": "putObject(String bucketName, String objectName, InputStream body, long size, String contentType)",
  "args": "contentType: application/javascript",
  "duration": 8825,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115), java.net.SocketOutputStream.write(SocketOutputStream.java:155), okio.Okio$1.write(Okio.java:78), okio.AsyncTimeout$1.write(AsyncTimeout.java:179), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:41), okhttp3.internal.http1.Http1Codec$FixedLengthSink.write(Http1Codec.java:286), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:171), okio.RealBufferedSink.write(RealBufferedSink.java:109), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:74), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:62), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185), okhttp3.RealCall.execute(RealCall.java:69), io.minio.MinioClient.executeReq(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executePut(MinioClient.java:1399), io.minio.MinioClient.executePut(MinioClient.java:1421), io.minio.MinioClient.putObject(MinioClient.java:3986), io.minio.MinioClient.putObject(MinioClient.java:4078), io.minio.MinioClient.putObject(MinioClient.java:3952), FunctionalTest.testPutObjectUnknownStreamSize(FunctionalTest.java:556), FunctionalTest.putObject_test8(FunctionalTest.java:582), FunctionalTest.runTests(FunctionalTest.java:2998), FunctionalTest.main(FunctionalTest.java:3154)]"
}

Executed 8 out of 13 tests successfully.
```

**Pull request merged**